src/Dropdown/Dropdown.js

```diff
@@ -121,45 +120,42 @@ export default class Dropdown extends React.Component {
     }
   };
 
-  attachAutocloseListener() {
-    if (!this.props.autoclose) {
-      return;
-    }
-    document.addEventListener('click', this.autoclose);
-  }
-
-  detachAutocloseListener() {
-    document.removeEventListener('click', this.autoclose);
-  }
-
   autoclose = e => {
-    if (this.state.isOpen && !this.menuRef.current.contains(e.target)) {
+    if (this.state.isOpen) {
       this.hide();
     }
   };
 
+  handleBlur = () => {
+    this.timeoutId = setTimeout(this.autoclose, 500);
+  }
+
+  handleFocus = () => {
+    clearTimeout(this.timeoutId);
+  }
+
   render() {
     const { width, trigger, render, children } = this.props;
     const { isOpen } = this.state;
 
     const renderFn = render || children;
 
     return (
-      <Fragment>
-        <DropdownTrigger ref={this.triggerRef} aria-haspopup="true" aria-expanded={isOpen}>
+      <DropdownWrapper tabIndex={0} onBlur={this.handleBlur}>
+        <DropdownTrigger onClick={this.toggle} ref={this.triggerRef} aria-haspopup="true" aria-expanded={isOpen}>
```

Comment:
Do we want to clone the `trigger` and pass it these props instead ([example](https://stackoverflow.com/a/32371612)) so we can avoid using this wrapper div?

Reply:
Sure - if it continues to work as intended. There may be an issue with passing the ref to `trigger`

Reply:
Actually since trigger is a JSX template literal (and not an element) this is best left as is.